pkg/codegen/pcl/binder_test.go

```diff
@@ -736,6 +736,190 @@ resource "randomPet" "random:index/randomPet:RandomPet" {
 	assert.NotNil(t, program)
 }
 
+// Tests that valid applications of the `call` intrinsic with simple receivers bind with no errors or diagnostics.
+func TestCallValidSimpleReceiver(t *testing.T) {
+	t.Parallel()
+
+	// Arrange.
+	source := `
+resource "c" "component:index:Callable" {
+	value = "bar"
+}
+
+cRes1 = call(c, "identity", {})
+cRes2 = call(c, "prefixed", { prefix = "foo-" })
+`
+
+	// Act.
+	program, diags, err := ParseAndBindProgram(t, source, "program.pp")
+
+	// Assert.
+	require.NoError(t, err)
+	require.Empty(t, diags, "There are no error or warning diagnostics")
+	require.NotNil(t, program)
+}
+
+// Tests that valid applications of the `call` intrinsic with complex receivers (e.g. compound expressions that
+// eventually result in resources) bind with no errors or diagnostics.
+func TestCallValidComplexReceiver(t *testing.T) {
+	t.Parallel()
+
+	// Arrange.
+	source := `
+resource "c" "component:index:Callable" {
+	value = "bar"
+}
+
+x = { c = c }
+
+cRes1 = call(x.c, "identity", {})
+cRes2 = call(x.c, "prefixed", { prefix = "foo-" })
+
+resource "cs" "component:index:Callable" {
+	options { range = 2 }
+	value = "quux"
+}
+
+csRes1 = call(cs[0], "identity", {})
+csRes2 = call(cs[1], "prefixed", { prefix = "baz-" })
+`
+
+	// Act.
+	program, diags, err := ParseAndBindProgram(t, source, "program.pp")
+
+	// Assert.
+	require.NoError(t, err)
+	require.Empty(t, diags, "There are no error or warning diagnostics")
+	require.NotNil(t, program)
+}
+
+// Tests that applications of the `call` intrinsic with no arguments fail to bind with appropriate diagnostics.
+func TestCallNoArguments(t *testing.T) {
+	t.Parallel()
+
+	// Arrange.
+	source := `
+resource "c" "component:index:Callable" {
+	value = "bar"
+}
+
+cRes1 = call()
+`
+
+	// Act.
+	program, diags, err := ParseAndBindProgram(t, source, "program.pp")
+
+	// Assert.
+	require.ErrorContains(t, err, "call must be passed a receiver")
+	require.Len(t, diags, 4)
+	require.Contains(t, diags[0].Summary, "call must be passed a receiver")
```

Comment:
Maybe expand this to the full message for < 3 args, since I found it a bit confusing in the later tests where we do pass a receiver, but the other parts are missing.

```suggestion
	require.Contains(t, diags[0].Summary, "call must be passed a receiver, method name, and arguments object")
```